native/core/src/execution/planner.rs

```diff
@@ -922,13 +956,18 @@ impl PhysicalPlanner {
                 Ok(DataType::Decimal128(_p2, _s2)),
             ) => {
                 let data_type = return_type.map(to_arrow_datatype).unwrap();
+                let func_name = if options.is_integral_div {
```

Comment:
I just realized, we maybe able to reuse the previous case-match instead of here. 
We needed to treat `decimal_div` differently because we had to deal with rounding. However we just need to round down for IntegralDivide?
 I.e. instead of 77digits for scale, we only need 76digits that fits into Decimal256

Will need a similar calculation to
```
                || (op == DataFusionOperator::Modulo
                    && max(s1, s2) as u8 + max(p1 - s1 as u8, p2 - s2 as u8)
                        > DECIMAL128_MAX_PRECISION) 
```

In this way, we do not need the  `decimal_div` change?

Reply:
Thanks for your suggestion, it makes sense to me. But I am getting an overflow error and I will continue debugging this tomorrow.

![image](https://github.com/user-attachments/assets/fef0d1b4-c6be-4226-9e48-fc16c6103ab5)

Reply:
Thanks for trying, if this is too much trouble, we can file an issue ticket and can be worked on separately.

Reply:
This seems to be a bug in arrow, I have reported an issue: https://github.com/apache/arrow-rs/issues/7216

Reply:
> Thanks for trying, if this is too much trouble, we can file an issue ticket and can be worked on separately.

Could you please continue review this pr and let us keep changes to `decimal_div` ?

Reply:
Sure, perhaps it is good to mention the ticket in a comment
Otherwise, my only comment is

>Do we need to update https://github.com/apache/datafusion-comet/blob/main/spark/src/main/scala/org/apache/spark/sql/comet/DecimalPrecision.scala
?

> Additionally https://github.com/apache/datafusion-comet/blob/main/spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala#L1793 Decimal random number tests is another good test to extend

Reply:
Sorry I missed it before.

> Do we need to update https://github.com/apache/datafusion-comet/blob/main/spark/src/main/scala/org/apache/spark/sql/comet/DecimalPrecision.scala
?

I guess it may not be necessary, IntegralDivide always returns a long type.

>Additionally https://github.com/apache/datafusion-comet/blob/main/spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala#L1793 Decimal random number tests is another good test to extend

I added div operator to this test case.